R/update_hub_target_data.R

```diff
@@ -0,0 +1,124 @@
+nhsn_col_names <- list(
+  covid = "totalconfc19newadm",
+  rsv = "totalconfrsvnewadm"
+)
+nssp_col_names <- list(
+  covid = "percent_visits_covid",
+  rsv = "percent_visits_rsv"
+)
+
+##' Get and process target data for the hub
+##'
+##' This function pulls, formats and save NHSN and NSSP data for use in the hub.
+##'
+##' @param base_hub_path Path to the base hub directory.
+##' @param disease Disease name ("covid" or "rsv").
+##' @param excluded_locations Vector of location codes to exclude from the output.
+##' Default value is c("78", "69", "66", "60"), corresponding to "US Virgin Islands",
+##' "Guam", "American Samoa", and "Northern Mariana Islands".
+##' @param nhsn_first_weekending_date First week-ending date to include for
+##' the NHSN dataset. Default value is "2024-11-09".
+##' @param legacy_file Logical. Whether to write legacy CSV output (default: FALSE).
+##'
+##' @return Writes `time-series.parquet` and optionally legacy csv target data files
+##' to the target-data directory in the hub.
+##' @export
+update_hub_target_data <- function(
+  base_hub_path,
+  disease,
+  nhsn_first_weekending_date = lubridate::as_date("2024-11-09"),
+  excluded_locations = c("78", "69", "66", "60"),
+  legacy_file = FALSE
+) {
+  if (!disease %in% c("covid", "rsv")) {
+    stop("'disease' must be either 'covid' or 'rsv'")
+  }
+  today <- lubridate::today()
+  output_dirpath <- fs::path(base_hub_path, "target-data")
+  fs::dir_create(output_dirpath)
+
+  nhsn_col_name <- nhsn_col_names[[disease]]
+  nssp_col_name <- nssp_col_names[[disease]]
+
+  nhsn_data <- forecasttools::pull_data_cdc_gov_dataset(
+    dataset = "nhsn_hrd_prelim",
+    columns = nhsn_col_name,
+    start_date = nhsn_first_weekending_date
+  ) |>
+    dplyr::rename(
+      observation = nhsn_col_name,
+      date = "weekendingdate"
+    ) |>
+    dplyr::mutate(
+      date = lubridate::as_date(.data$date),
+      observation = as.numeric(.data$observation),
+      jurisdiction = stringr::str_replace(.data$jurisdiction, "USA", "US")
+    ) |>
+    dplyr::mutate(
+      location = forecasttools::us_location_recode(
+        .data$jurisdiction,
+        "abbr",
+        "code"
+      ),
+      as_of = !!today,
+      target = glue::glue("wk inc {disease} hosp")
+    ) |>
```

Comment:
I believe this pipechain can be simplified somewhat (though you'll have to change the select and rename calls below slightly.
```suggestion
    dplyr::mutate(
      date = lubridate::as_date(.data$weekendingdate),
      observation = as.numeric(.data[[nhsn_col_name]]),
      location = stringr::str_replace(.data$jurisdiction, "USA", "US") |> 
          forecasttools::us_location_recode("abbr", "code"),
       as_of = !!today,
       target = glue::glue("wk inc {disease} hosp")
       ) |>
```